profile-widget/diveprofileitem.cpp

```diff
@@ -618,45 +620,47 @@ void DiveGasPressureItem::replot(const dive *d, int fromIn, int toIn, bool in_pl
 	// Right now it's just strictly alternating when you have multiple gas
 	// pressures.
 
-	QFlags<Qt::AlignmentFlag> alignVar = Qt::AlignTop;
-	std::vector<QFlags<Qt::AlignmentFlag>> align(pInfo.nr_cylinders);
-
-	double labelHeight = DiveTextItem::fontHeight(dpr, 1.0);
+	QFlags<Qt::AlignmentFlag> startAlignVar = Qt::AlignTop;
 
 	for (const Segment &segment: segments) {
 		// Magic Y offset depending on whether we're aliging
 		// the top of the text or the bottom of the text to
 		// the pressure line.
-		double value_y_offset = -0.5 * dpr;
-		double label_y_offset = alignVar & Qt::AlignTop ? labelHeight : -labelHeight;
-		gasmix gas = get_cylinder(d, segment.cyl)->gasmix;
-		plotPressureValue(segment.first.pressure, segment.first.time, alignVar, value_y_offset);
-		plotGasValue(segment.first.pressure, segment.first.time, gas, alignVar, label_y_offset);
+		double y_offset = -0.5 * dpr;
+		plotPressureValue(segment.first.pressure, segment.first.time, startAlignVar, y_offset);
 
 		// For each cylinder, on right hand side of the curve, write cylinder pressure
-		plotPressureValue(segment.last.pressure, segment.last.time, alignVar | Qt::AlignLeft, value_y_offset);
+		double x_offset = plotPressureValue(segment.last.pressure, segment.last.time, Qt::AlignTop | Qt::AlignLeft, y_offset) + 2;
+		plotGasValue(segment.last.pressure, segment.last.time, get_cylinder(d, segment.cyl), Qt::AlignTop | Qt::AlignLeft, x_offset, y_offset, showDescriptions);
 
 		/* Alternate alignment as we see cylinder use.. */
-		alignVar ^= Qt::AlignTop | Qt::AlignBottom;
+		startAlignVar ^= Qt::AlignTop | Qt::AlignBottom;
 	}
 }
 
-void DiveGasPressureItem::plotPressureValue(double mbar, double sec, QFlags<Qt::AlignmentFlag> align, double pressure_offset)
+double DiveGasPressureItem::plotPressureValue(double mbar, double sec, QFlags<Qt::AlignmentFlag> align, double y_offset)
 {
 	const char *unit;
-	int pressure = get_pressure_units(lrint(mbar), &unit);
+	auto label = QString("%1%2").arg(get_pressure_units(lrint(mbar), &unit)).arg(unit);
```

Comment:
I know that you only moved code here and not that it matters at all, but the 'pure' thing to do is `QStringLiteral()`, as the format string will then not be constructed at runtime.

That said, the more you work on the profile, the more it becomes a pressing issue to get a resolution for #3863. Rebasing is becoming very painful and error-prone. :(

Reply:
PS: of course the biggest pain is self-inflicted with my porting to C++.

Reply:
@bstoeger: Yes, I suspect that #3863 will be a rebase from hell, mostly due to changes that I made. But a lot of these changes were motivated by bug reports or shortcomings reported by users, and I don't think holding off this kind of thing for months on end is helpful if we want to keep our users engaged with helping us by providing feedback.
Is there anything that can be done to unblock you on #3863?

Reply:
> Yes, I suspect that https://github.com/subsurface/subsurface/pull/3863 will be a rebase from hell, mostly due to changes that I made.

No worries, the worst problem will be my own conversions to C++.

I'm stuck with the link-order issues on mobile. The only fix I can think about is moving all the chart/QtQuick related stuff in a single "librabry". :( Then needs of course testing on mobile, with pinch-zooming, etc.

Currently, conversion to C++ has priority for me, though.